Adding SSA observer (#1033)
ihmeuw/vivarium_research: docs/source/concept_models/vivarium_census_synthdata/concept_model.rst

#. Currently mailing addresses are the same as home addresses. This is not true, especially for rural populations. We plan to add this to the model later. 

Social Security Observer
^^^^^^^^^^^^^^^^^^^^^^^^

Social security information is kept and tracked for a range 
of different actions. For simplicity, we will limit this 
section to only creation and dates of death. Others could be 

> added later, if desired. 

Reviewer: Let's be clear here that "later" = not Model 1/the initial full-US run, if that is what we mean.

Author: Added

**When to Sample** 

- A sample of qualifying events is taken on every time step 

**What to Sample** 

.. list-table:: Simulant Attributes to Sample 
  :widths: 20
  :header-rows: 0

  * - Unique simulant ID (for PRL tracking)
  * - First name
  * - Middle initial 
  * - Last name
  * - DOB 
  * - Social Security Number 
  * - Type of event 
  * - Date of event 

Currently, we will only track 2 types of events: 

#. Creation of a SSN 
#. Date of death recorded 

The creation of a SSN can be triggered by a birth or by immigration into 
the US. Both will be listed as "Creation" and are not differentiated. This 
event is only recorded for simulates that receive a SSN. 
The date of event is either the date of birth or the date of immigration. 

> If these values are unknown, any date within the time step can be recorded. 

Reviewer: Fine to have this here, but my understanding is that these dates will always be known.

Author: Okay! I wasn't sure if simulants got an immigration "date" or just that they moved on a time step. If they get a date then I can delete this.

The date of death is a recording of anyone who has died and has a SSN. The 
date of the event is the date of death. This will be listed as "Date 
of Death". 

Here is an example: 

> .. image:: SSA_example.png

Reviewer: In general/for the future, it would be nice to have these as actual tables instead of images. But, no big deal if that is a pain to do here.

Author: Okay, I can switch them in the future. Since they are used throughout this section and match I'm going to leave them all for now.

Note that the top row is a simulate that received a SSN at birth. The 
third row indicates someone who immigrated at age 5 and received their 
SSN then. 

**Who to Sample** 

100% of simulants with a SSN and a qualifying event in that time step 
will be recorded. 

> **Data Errors/Noise** 

Reviewer: From the SE perspective, the fact that this section is "parameterizable" means "you should implement the rest of component, but we plan to change this part." For that reason, it would make sense to me to put the bullets in this section in `..todo` blocks or similar to make them stand out.

Author: Sure, added these in a todo block

- The omission rate (currently 0%) should be parameterizable. 

> - Additional noise in names and dates will be added in a separate function, at a later date. 

Reviewer: ```suggestion
- Additional noise functions on names and dates should be parameterizable.
```

> **Limitations and Possible Future Adds** 

Reviewer: I think this could be a bit confusing -- because "possible future adds" doesn't sound so different from saying that something is parameterizable.

I'd just say "Limitations," or if you want to keep the second part, somehow make it clear that this means in a future model, not the minimum viable model we will be using for our first US-scale run.

Author: Okay, makes sense. I changed the language to be "Items NOT Included in Minimum Viable Model" which I think is more clear.

There are a significant number of other possible "events" that 
could be included in the observer. These are either not implemented 
in the larger simulation at this time, or more difficult and so 
have not been included. These are: 

#. Name changes - either first names (trans folks or others) or last names (commonly marriage or divorce). Not included in larger simulation 
#. Sex-coding changes - not included in larger simulation 

> #. Correction of incorrect information - not included in larger simulation 

Reviewer: "Not included in larger simulation" doesn't seem like the right way to say this, since this event doesn't mean much outside the context of this observer.

By the way, this list of other events is a good example of something that I think would be infeasible or very difficult to do outside the sim (cannot be parameterizable), which is why it's especially important that we are clear that these are not happening in our Model 1 (see comment above).

Author: Okay, yeah, makes sense. I think this will be a tough add because to track a "correction" we first need to intentionally get something wrong and "remember" that error. I wrote a bit more to express this more fully in the note.

> #. Social security benefits - this could be an easy add here if we add it in employment 

Reviewer: ```suggestion
#. Receipt of social security benefits - this could be an easy add here if we add it in employment 
```

Author: Thanks!

> #. Diability benefits - similarly, could be an easy add if it is added in employment 

Reviewer: ```suggestion
#. Receipt of disability benefits - similarly, could be an easy add if it is added in employment 
```

> #. New employment - it's unclear what percent of new jobs are actually recorded. Also this leads to complications if someone shares a SSN. 

Reviewer: Right now in the sim, nobody shares an SSN.
```suggestion
#. New employment - it's unclear what percent of new jobs are actually recorded. Also this could be harder in a future model that included shared SSNs. 
```

Author: We actually do have this in the tax observer! So while no simulants share a SSN in the state table, they actually can "borrow" a SSN in an observer. I have clarified that here.
